x/ccv/consumer/keeper/validators.go

```diff
@@ -89,16 +108,32 @@ func (k Keeper) ValidatorByConsAddr(sdk.Context, sdk.ConsAddress) stakingtypes.V
 		Also, the slashing module will cal lthis function when it observes downtime. In that case
 		the only requirement on the returned value is that it isn't null.
 	*/
+	if k.IsPreCCV(ctx) || ctx.BlockHeight() <= k.LastSovereignHeight(ctx) {
+		if k.stakingKeeper == nil {
+			return stakingtypes.Validator{}
+		}
+
+		return k.stakingKeeper.ValidatorByConsAddr(ctx, consAddr)
+	}
 	return stakingtypes.Validator{}
 }
 
 // Slash queues a slashing request for the the provider chain
 // All queued slashing requests will be cleared in EndBlock
-func (k Keeper) Slash(ctx sdk.Context, addr sdk.ConsAddress, infractionHeight, power int64, _ sdk.Dec, infraction stakingtypes.InfractionType) {
+func (k Keeper) Slash(ctx sdk.Context, addr sdk.ConsAddress, infractionHeight, power int64, slashFactor sdk.Dec, infraction stakingtypes.InfractionType) {
 	if infraction == stakingtypes.InfractionEmpty {
 		return
 	}
 
+	if k.IsPreCCV(ctx) || ctx.BlockHeight() <= k.LastSovereignHeight(ctx) {
```

Comment:
Can never be the case that `ctx.BlockHeight() < k.LastSovereignHeight(ctx)` as the CCV module was not on the chain.

Reply:
This is not the correct way to deal with slashing. This slash request could be for one of the following cases:
- An infraction committed by a sovereign chain validator. This can be detected via the infraction height. The sovereign chain validator set is active until the height `h` (inclusive). Note that `h != LastSovereignHeight`. `LastSovereignHeight` is the height the provider validator set was sent to TM. According to the [ABCI spec](https://github.com/cometbft/cometbft/blob/v0.34.27/spec/abci/abci.md#endblock),  `h = LastSovereignHeight + 1`. In this case, the validator should be slashed if the unbonding period didn't pass yet. **Still need to implement a way in democracy-staking that not slash validators after unbonding period elapsed after the switch**.
-